packages/gasket-plugin-lifecycle/index.js

```diff
@@ -26,8 +26,9 @@ async function resolve(root, name) {
     }
   }
 
+  const reg = /.*^((?!(\.test|\.spec).js).)*\.js$/;
```

Comment:
It's odd that you have a `^` anchor _after_ a `.*`, meaning that `.*` is never necessary. This should be equivalent:

```js
const isJs = /\.js$/i;
const isTest = /\.(spec|test)\./i;

return files
  .filter(file => isJs.test(file) && !isTest.test(file))
  .map(...)
```

Reply:
Much simpler. Fixed